src/main/java/core/basesyntax/DataReaderImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataReaderImpl implements DataReader {
+    private final FileReader fileReader;
+
+    public DataReaderImpl(FileReader fileReader) {
+        this.fileReader = fileReader;
+    }
+
+    @Override
+    public List<String> readFromFile(String path) {
+        List<String> lines = new ArrayList<>();
+
+        try (BufferedReader br = new BufferedReader(fileReader)) {

```

Comment:
Instead of using the class-level `fileReader`, create a new `FileReader` instance here using the provided `path` parameter: `try (BufferedReader br = new BufferedReader(new FileReader(path))) {`.